app/app.rb

```diff
@@ -0,0 +1,104 @@
+ENV['RACK_ENV'] ||= 'development'
+
+require 'sinatra/base'
+require 'sinatra/flash'
+require './app/datamapper_setup'
+require_relative 'app_helper'
+
+class Chitter < Sinatra::Base
+  enable :sessions
+  set :session_secret, 'key'
+  register Sinatra::Flash
+
+  get '/' do
+    redirect '/peeps'
+  end
+
+  get '/peeps' do
+    @peeps = Peep.all
+    @current_user = current_user
+    erb :peeps
+  end
+
+  get '/peeps/new' do
+    @current_user = current_user
+    if @current_user
+      erb :new_peep
+    else
+      redirect '/peeps'
+    end
+  end
+
+  get "/goto/peeps/new" do
+    redirect '/peeps/new'
+  end
+
+  post '/peeps/new' do
+    peep_msg = params[:peep]
+    timestamp = Time.new.strftime("%H:%M:%S %a-%d-%b-%Y")
+    new_peep = Peep.new(message: peep_msg, username: current_user.username,
+      time: timestamp)
+    current_user.peep << new_peep
+    current_user.save
+    redirect '/peeps'
+  end
+
+  get '/users/sign_in' do
+    erb :sign_in
+  end
+
+  get '/goto/users/sign_in' do
+    redirect '/users/sign_in'
+  end
+
+  post '/users/sign_in' do
+    login_name = params[:username_or_email]
+    password = params[:password]
+    user = find_user(login_name)
+    if user
```

Comment:
Whenever I see nested 'if' statements, I think that there must be a way to refactor the method. As a rule of thumb, if you have multiple levels of nesting, some of the logic probably belongs elsewhere.